sklearn/utils/_array_api.py

```diff
@@ -452,6 +458,34 @@ def isdtype(self, dtype, kind):
     def pow(self, x1, x2):
         return numpy.power(x1, x2)
 
+    def cumulative_sum(self, x, axis=None, dtype=None, include_initial=False):
+        # This is copy pasted from array-api-compat
```

Comment:
Is this required to support older versions of numpy?

Reply:
Yeah, cumulative_sum is new in numpy 2.1.

Reply:
I am not sure about that. Maybe @ogrisel or @betatim would have a better idea.

Reply:
Is `cumulative_sum` not in array api? https://data-apis.org/array-api/latest/API_specification/generated/array_api.cumulative_sum.html#array_api.cumulative_sum

Reply:
It is. 
The issue happens though array API is turned off (and array-api-compat is not installed), since we need to rely on what's in numpy/what we have wrapped in the ``_NumpyAPIWrapper`` then.

(I could be misunderstanding how NumpyAPIWrappers is supposed to function though.)

Reply:
Ah good point, they should add a version added note in their docs!

I actually also have this problem over in #29431

Reply:
I think the PR for vendoring the array-api-compat and array-api-extra is close to being merged. So let's wait for that.